packages/cozy-procedures/src/components/overview/helpers.js

```diff
@@ -62,20 +62,22 @@ export const fetchBanksSettings = async client => {
 }
 
 export const fetchTransactionsHistoryContent = async client => {
+  const settingsPromise = fetchBanksSettings(client)
   const accounts = await fetchCheckingsLikeAccounts(client)
   const accountsIds = accounts.map(account => account._id)
   const transactions = await fetchTransactionsHistory(client, accountsIds)
-  const settings = await fetchBanksSettings(client)
+  const transactionsByAccount = groupBy(
+    transactions,
+    transaction => transaction.account
+  )
+  const settings = await settingsPromise
```

Comment:
IMHO it is a bit surprising to see this await here. I think the parallelizing intention would be clearer with a Promise.all.

```js
const fetchCheckingsLikeTransactions = async client => {
  const accounts = await fetchCheckingsLikeAccounts(client)
  const accountsIds = accounts.map(account => account._id)
  return fetchTransactionsHistory(client, accountsIds)
}

const [transactions, settings] = await Promise.all([
  fetchCheckingsTransactions(client),
  fetchBanksSettings(client),
])
```

Reply:
I wanted to leave the `groupBy` parallelized but you're right, I might be micro-optimizing and readability is more important.

Reply:
Is it better for you now?

Reply:
:+1: